lib/resource-timing.js

```diff
@@ -0,0 +1,228 @@
+        b: ["domInteractive", "domContentLoadedEventStart", "domContentLoadedEventEnd", "domComplete", "loadEventStart", "loadEventEnd"]
+      };
+
+      this.initialize();
+    },
+    SNPRTProto = SurfNPerfRT.prototype;
+
+  SNPRTProto._inList = function(origin, options) {
+    if (options.hasOwnProperty("whitelist")) {
+      if (options.whitelist.indexOf(origin) === -1) {
+        return false;
+      } else {
+        return true;
+      }
+    } else if (options.hasOwnProperty("blacklist")) {
+      if (options.blacklist.indexOf(origin) === -1) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
+  SNPRTProto.getOrigins = function(options) {
+    if(window.performance.getEntriesByType) {
+      var resources = window.performance.getEntriesByType("resource");
+      var origins = [];
+      for var resource in resources {
+        var url = new URL(resources[resource].name);
+        if (origins.indexOf(url.origin) === -1 && this._inList(url.origin, options)) {
+          origins.push(url.origin);
+        }
+      }
+
+      return origins;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto.getResourcesFromOrigin = function(origin) {
+    if(window.performance.getEntriesByType) {
+      var resources = window.performance.getEntriesByType("resource");
+      var resourcesFromOrigin = [];
+      for var resource in resources {
+        var url = new URL(resources[resource].name);
+        if (url.origin === origin) {
+          resourcesFromOrigin.push(resources[resource]);
+        }
+      }
+      return resourcesFromOrigin;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto.isBufferFull = function() {
+    if(window.performance.getEntriesByType) {
+      if(performance.onresourcetimingbufferfull) {
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto._nameExtend = function(name) {
+    if(name.charAt(0) == '/') {
+      return "http://" + window.location.hostname + name;
+    } else {
+      return name;
+    }
+  };
+
+  SNPRTProto.getResource = function(name) {
+    name = this._nameExtend(name);
+
+    return window.performance.getEntriesByName(name, "resource")[0]
+  };
+
+  SNPRTProto.duration = function(name, eventA, eventB, options) {
+    name = this._nameExtend(name);
+    var resource = this.getResource(name);
+    if (resource) {
+      if (resource[eventA] === 0 || resource[eventB] === 0) {
+        return false;
+      }
+      return surfnperf._roundedDuration(resource[eventB], resource[eventA], options);
+    } else {
+      return undefined;
+    }
+  };
+
+  SNPRTProto.start = function(name, options) {
+    name = this._nameExtend(name);
+
+    var resource = this.getResource(name);
+    if(resource == undefined) {
```

Comment:
`undefined` is returned by default in JS if the function logic doesn't hit a `return` statement, so this whole function can be simplified to:
```js
    var resource = this.getResource(name);
    if(resource) {
      surfnperf._round(resource.startTime, options);
    }
```